core/src/main/java/org/apache/iceberg/MergingSnapshotProducer.java

```diff
@@ -414,6 +423,10 @@ private void validateNoNewDeletesForDataFiles(TableMetadata base, Long startingS
         ValidationException.check(
             Arrays.stream(deleteFiles).noneMatch(deleteFile -> deleteFile.content() == FileContent.POSITION_DELETES),
             "Cannot commit, found new position delete for replaced data file: %s", dataFile);
+      }  else if (ignorePosDeletes) {
```

Comment:
Nit: extra space.

This logic looks incorrect to me. If equality deletes are not ignored, then this will always check for position deletes, regardless of the value for `ignorePosDeletes`.

Reply:
Nice catch! Let me break `if/else if/else`.